src/azure-cli/azure/cli/command_modules/synapse/commands.py

```diff
@@ -0,0 +1,75 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def load_command_table(self, _):
+    from azure.cli.core.commands import CliCommandType
+    from azure.cli.core.profiles import ResourceType
+    from ._client_factory import cf_synapse_client_workspace_factory
+    from ._client_factory import cf_synapse_client_bigdatapool_factory
+    from ._client_factory import cf_synapse_spark_batch
+    from ._client_factory import cf_synapse_spark_session
+
+    synapse_workspace_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.synapse.operations#WorkspacesOperations.{}',
+        client_factory=cf_synapse_client_workspace_factory,
+        resource_type=ResourceType.MGMT_SYNAPSE
+    )
+
+    synapse_bigdatapool_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.synapse.operations#BigDataPoolsOperations.{}',
+        client_factory=cf_synapse_client_bigdatapool_factory,
+        resource_type=ResourceType.MGMT_SYNAPSE
+    )
+
+    synapse_spark_batch_sdk = CliCommandType(
+        operations_tmpl='azure.synapse.operations#SparkBatchOperations.{}',
+        client_factory=cf_synapse_spark_batch
+    )
+
+    synapse_spark_session_sdk = CliCommandType(
+        operations_tmpl='azure.synapse.operations#SparkSessionOperations.{}',
+        client_factory=cf_synapse_spark_session
+    )
+
+    # Management Plane Commands
+    with self.command_group('synapse workspace', synapse_workspace_sdk,
+                            client_factory=cf_synapse_client_workspace_factory) as g:
+        g.show_command('show', 'get')
+        g.custom_command('list', 'list_workspaces')
+        g.custom_command('create', 'create_workspace', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
```

Comment:
Add a "wait" command as you use supports_no_wait for create and delete commands. [https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md)

Reply:
> Add a "wait" command as you use supports_no_wait for create and delete commands. https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md

Added